frontend_tests/casper_tests/05-subscriptions.js

```diff
@@ -140,11 +140,19 @@ casper.then(function () {
         casper.click('form#stream_creation_form button.btn.btn-default');
         casper.fill('form#add_new_subscription', {stream_name: '  '});
         casper.click('#create_stream_button');
-        casper.fill('form#stream_creation_form', {stream_name: 'Waseemio'});
+        casper.fill('form#stream_creation_form', {stream_name: 'Waseemio@'});
         casper.click('form#stream_creation_form button.btn.btn-primary');
     });
 });
 casper.then(function () {
+   casper.waitForSelectorText('#stream_name_error', 'A stream needs to have valid characters', function () {
+       casper.test.assertTextExists('A stream needs to have valid characters', "Can't create a stream with invalid characters");
+        casper.click('#create_stream_button');
+        casper.fill('form#stream_creation_form', {stream_name: 'Waseemio'});
```

Comment:
You probably need to wait for 'form#stream_creation_form' here.

Reply:
@umairwaheed I have written these tests by following the same pattern of the existing tests. In the existing tests, we were not waiting for` form#stream_creation_form`. Also all the Travis frontend test were passed.

Reply:
Adding a wait will make your test more deterministic. If you don't add a wait, your test will pass most of the time, but since this is an asynchronous operation, there is a chance that the control reaches this statement before the form exists.

Reply:
@umairwaheed Okaay thankyou! That makes sense. I'll make the changes.